File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int n=s.length(),m=t.length();
        if(n!=m) return false;
        map<char,int>mp1;
        map<char,int>mp2;

        for(int i=0;i<n;i++){
            if(mp1[s[i]]!=mp2[t[i]]) return false;
            mp1[s[i]] = i+1;
            mp2[t[i]] = i+1;

        }
        return true;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int n = s.length(), m= t.length();
        if(n!=m) return false;
        Map<Character, Character> mps = new HashMap<>();
        Map<Character, Character> mpt = new HashMap<>();
        for(int i=0;i<n;i++) {
            char charS = s.charAt(i);
            char charT = t.charAt(i);
            
            if(mps.containsKey(charS)) {
                if(mps.get(charS)!=charT) return false;
            }else {
                mps.put(charS,charT);
            }
            
            if(mpt.containsKey(charT)) {
                if(mpt.get(charT)!=charS) return false;
            }else {
                mpt.put(charT, charS);
            }
        }
        return true;
    }
}